### ISSUE TITLE
Register Shiny JavaScript methods earlier in the life-cycle

### ISSUE THREAD

**Issue by cpsievert** [rstudio/shiny#2655]
Some of Shiny's JavaScript methods (e.g., `Shiny.setInputValue`) are registered when `initShiny` executes, which is scheduled here:

https://github.com/rstudio/shiny/blob/89bd7e90117b5ed0c99987d81c8ec8b51a3305c1/srcjs/init_shiny.js#L511-L515

This implies that, when htmlwidget(s) are statically rendered in Shiny, `HTMLWidgets.staticRender()` executes one tick before `initShiny()`, which means those JS methods aren't available to the widgets `renderValue()` method. Some widgets, like leaflet, are privy to this and schedule their methods (which reference Shiny's methods) to be called one tick later, but widget authors shouldn't be expected to know this:

https://github.com/rstudio/leaflet/blob/18ec9817cf3261ed3cc0aa5d0d030d6c36ba0667/javascript/src/index.js#L266-L268

In other words, instead of delaying the method registration that happens `initShiny`, it should probably happen right away (when Shiny is loaded)
